File: src/main/java/com/mxmariner/andxtidelib/remote/RemoteStation.java
package com.mxmariner.andxtidelib.remote;

import android.os.Parcel;
import android.os.Parcelable;
import android.os.RemoteException;

public class RemoteStation implements Parcelable {
    private final long stationId;
    private final double latitude;
    private final double longitude;
    private final StationType type;

    public RemoteStation(long stationId, double latitude, double longitude, StationType type) {
        this.stationId = stationId;
        this.latitude = latitude;
        this.longitude = longitude;
        this.type = type;
    }

    public long getStationId() throws RemoteException {
        return stationId;
    }

    public double getLatitude() throws RemoteException {
        return latitude;
    }

    public double getLongitude() throws RemoteException {
        return longitude;
    }

    protected RemoteStation(Parcel in) {
        stationId = in.readLong();
        latitude = in.readDouble();
        longitude = in.readDouble();
        type = in.readParcelable(StationType.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(stationId);
        dest.writeDouble(latitude);
        dest.writeDouble(longitude);
        dest.writeParcelable(type, flags);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<RemoteStation> CREATOR = new Parcelable.Creator<RemoteStation>() {
        @Override
        public RemoteStation createFromParcel(Parcel in) {
            return new RemoteStation(in);
        }

        @Override
        public RemoteStation[] newArray(int size) {
            return new RemoteStation[size];
        }
    };
}

File: mxtide/src/main/java/com/mxmariner/andxtidelib/remote/RemoteStation.kt
package com.mxmariner.andxtidelib.remote

import android.os.Parcel
import android.os.Parcelable

class RemoteStation(val stationId: Long,
                    val latitude: Double,
                    val longitude: Double,
                    val type: StationType) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readLong(),
            parcel.readDouble(),
            parcel.readDouble(),
            parcel.readParcelable(StationType::class.java.classLoader))

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(stationId)
        parcel.writeDouble(latitude)
        parcel.writeDouble(longitude)
        parcel.writeParcelable(type, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<RemoteStation> {
        override fun createFromParcel(parcel: Parcel): RemoteStation {
            return RemoteStation(parcel)
        }

        override fun newArray(size: Int): Array<RemoteStation?> {
            return arrayOfNulls(size)
        }
    }
}
